### ISSUE TITLE
Get ready for `@docImport`s

### ISSUE THREAD

**Issue by goderbauer** [flutter/flutter#150800]

**Comment by goderbauer** [comment]
To get some numbers: Before we started there where 7,631 unresolved comment references in pkg:flutter and 8,260 in the entire repository.

**Comment by goderbauer** [comment]
We are down to 5,036 violations in pkg:flutter (mainly in the widgets, rendering, and material library) and 5,371 in the entire repository.

**Comment by LongCatIsLooong** [comment]
I'm seeing that the analyzer could not redirect `typedef`s (or maybe it's because the comment ref points to an enum value?), and the dart language server does not resolve those in my IDE either. An example: [MaterialState.focused](https://main-api.flutter.dev/flutter/widgets/WidgetState.html) in  https://main-api.flutter.dev/flutter/material/MaterialStateUnderlineInputBorder-class.html

where `MaterialWidgetState` a deprecated typedef that now points to `WidgetState`. The language server can resolve the prefix (`MaterialWidgetState`) just fine but fails to resolve [MaterialState.focused](https://main-api.flutter.dev/flutter/widgets/WidgetState.html)

**Comment by LongCatIsLooong** [comment]
It looks like the analyzer does not know about dartdoc macros, it won't be able to provide feedback if a macro expansion introduces out-of-scope comment reference. So it's unfortunate that those won't turn up until in presubmits.

**Comment by goderbauer** [comment]
cc @srawlins for the issues that @LongCatIsLooong found above.

**Comment by goderbauer** [comment]
After xxx lands we are down to just 104 violations in packages that are relevant for our published docs. Fixing all of these is blocked on resolving the unresolved blockers listed in the top comment of this issue.

**Comment by goderbauer** [comment]
I did another scan over the repository. After https://github.com/flutter/flutter/pull/160918 lands we have 67 issues remaining in packages that we publish docs for. Fixing those is blocked on the unchecked blockers listed in the top comment.

**Comment by goderbauer** [comment]
https://github.com/dart-lang/sdk/issues/56528 and https://github.com/dart-lang/sdk/issues/56527 are required fixes before we can enforce the `comment_references` lint.

https://github.com/dart-lang/sdk/issues/56109 is lower priority and we can work around that issue with two `// ignore`s for now.

**Comment by goderbauer** [comment]
For future references: https://github.com/flutter/flutter/pull/160921 is a PR that turns on the `comment_references` lint in the appropriate places. It currently fails CI because of the aforementioned issues.
